inc/customizer/controls/react/src/icon-picker/IconPicker.js

```diff
@@ -0,0 +1,70 @@
+import * as icons from '../common/icons';
+import { useState, useEffect } from '@wordpress/element';
+
+const IconPicker = ({ selectIcon, isOpen }) => {
+	const [socialIcons, setSocialIcons] = useState();
+	const [filtered, filterIcons] = useState();
```

Comment:
You shouldn't need to have state and a `useEffect` for this, you can simply use a variable like `filteredIcons` before the render where you filter them.